sdk/src/main/java/me/uport/sdk/Uport.kt

```diff
@@ -27,33 +24,16 @@ object Uport {
 
     private lateinit var prefs: SharedPreferences
 
-    private lateinit var accountCreator: AccountCreator
-
-    private var defaultAccountHandle = ""
-
-    var defaultAccount: Account?
-        get() = accountStorage?.get(defaultAccountHandle)
-        set(value) {
-            val newDefault = value?.copy(isDefault = true)
-            @Suppress("LiftReturnOrAssignment")
-            if (newDefault == null) {
-                accountStorage?.delete(defaultAccountHandle)
-                defaultAccountHandle = ""
-            } else {
-                val oldAccounts = accountStorage
-                        ?.all()
-                        ?.map { it.copy(isDefault = false) }
-                        ?: emptyList()
-                accountStorage?.upsertAll(oldAccounts + newDefault)
-                defaultAccountHandle = newDefault.handle
-            }
-        }
+    private lateinit var accountCreator: HDAccountCreator
 
-    private var accountStorage: AccountStorage? = null
+    private var accountStorage: SharedPrefsAccountStorage? = null
 
-    private const val UPORT_CONFIG: String = "uport_sdk_prefs"
+    @Suppress("UnsafeCast")
+    var defaultAccount: HDAccount?
+        get() = accountStorage?.getDefaultAccount() as HDAccount?
+        set(value) { }
```

Comment:
`set` should call into the account storage method

Reply:
What exactly should happen in `set` ?

Reply:
it should call the method in accountStorage that saves the default account.

I don't understand exactly what you are not understanding.

Having an empty `set()` method means that when you do `Uport.defaultAccount = something` nothing happens.. literally nothing
So one would have to call the accountStorage directly to actually set the default account.

Reply:
Alright, I will set it there as well but it does get set when the account is created.

Reply:
It becomes simpler if `Uport.defaultAccount = theNewAccount` is called after creation.
This way there is even more decoupling from the account storage and less duplication.